From 0000000000000000000000000000000000000000 Mon Sep 17 00:00:00 2001
From: fgei <fgei@gmail.com>
Date: Wed, 10 Jul 2024 13:47:10 +0000
Subject: [PATCH] tmp: config-generator: Disable conditional components

On Pixels with MTE support, crashes are encountered on calls with
parsing and allocating new arrays for conditional components. Disable
support temporarily for conditional components until this memory tagging
issue is further investigated and resolved properly.
---
 .../app/vanadium/config/host/ConfigGenerator.java   | 13 +++++++++++++
 1 file changed, 13 insertions(+)

diff --git a/vanadium/android_config/proto/java/src_host/app/vanadium/config/host/ConfigGenerator.java b/vanadium/android_config/proto/java/src_host/app/vanadium/config/host/ConfigGenerator.java
index 7f1c4cfbc9d95..1bdbe3a0793e7 100644
--- a/vanadium/android_config/proto/java/src_host/app/vanadium/config/host/ConfigGenerator.java
+++ b/vanadium/android_config/proto/java/src_host/app/vanadium/config/host/ConfigGenerator.java
@@ -71,6 +71,19 @@ public class ConfigGenerator {
                         )
                 ))
         ));
+        configList.add(config(configParams -> configParams.setSpec(
+                        spec(specParams -> specParams.setSpecTypes(getSpecTypes(
+                                SpecType.BROWSER
+                        ))
+                                .setMinVersion(6478_000_00L)
+                                .setMaxVersion(6478_179_99L)
+                        ))
+                .addAllFlags(flags(
+                        flag(flagParams -> flagParams.setFlagName("VanadiumConfCondtionalsSupport")
+                                .setFlagType(FlagType.DISABLED_FEATURE)
+                        )
+                ))
+        ));
         configList.add(config(configParams -> configParams.setSpec(
                         spec(specParams -> specParams.setSpecTypes(getSpecTypes(
                                 SpecType.BROWSER
